test/core/client_channel/resolvers/binder_resolver_test.cc

```diff
@@ -0,0 +1,194 @@
+
+    void ReturnResult(grpc_core::Resolver::Result result) override {
+      EXPECT_TRUE(expect_result_);
+      ASSERT_TRUE(result.addresses.size() == 1);
+      grpc_core::ServerAddress addr = result.addresses[0];
+      const struct sockaddr_un* un =
+          reinterpret_cast<const struct sockaddr_un*>(addr.address().addr);
+      EXPECT_EQ(addr.address().len,
+                sizeof(un->sun_family) + expected_binder_id_.length() + 1);
+      EXPECT_EQ(un->sun_family, AF_MAX);
+      EXPECT_EQ(un->sun_path, expected_binder_id_);
+    }
+
+    void ReturnError(grpc_error_handle error) override {
+      GRPC_ERROR_UNREF(error);
+    }
+
+   private:
+    // Whether we expect ReturnResult function to be invoked
+    bool expect_result_;
+
+    std::string expected_binder_id_;
+  };
+
+  void TestSucceeds(const char* string, const std::string& expected_path) {
+    gpr_log(GPR_DEBUG, "test: '%s' should be valid for '%s'", string,
+            factory_->scheme());
+    grpc_core::ExecCtx exec_ctx;
+    absl::StatusOr<grpc_core::URI> uri = grpc_core::URI::Parse(string);
+    ASSERT_TRUE(uri.ok()) << uri.status().ToString();
+    grpc_core::ResolverArgs args;
+    args.uri = std::move(*uri);
+    args.result_handler =
+        absl::make_unique<BinderResolverTest::ResultHandler>(expected_path);
+    grpc_core::OrphanablePtr<grpc_core::Resolver> resolver =
+        factory_->CreateResolver(std::move(args));
+    ASSERT_TRUE(resolver != nullptr);
+    resolver->StartLocked();
+  }
+
+  void TestFails(const char* string) {
+    gpr_log(GPR_DEBUG, "test: '%s' should be invalid for '%s'", string,
+            factory_->scheme());
+    grpc_core::ExecCtx exec_ctx;
+    absl::StatusOr<grpc_core::URI> uri = grpc_core::URI::Parse(string);
+    ASSERT_TRUE(uri.ok()) << uri.status().ToString();
+    grpc_core::ResolverArgs args;
+    args.uri = std::move(*uri);
+    args.result_handler =
+        absl::make_unique<BinderResolverTest::ResultHandler>();
+    grpc_core::OrphanablePtr<grpc_core::Resolver> resolver =
+        factory_->CreateResolver(std::move(args));
+    EXPECT_TRUE(resolver == nullptr);
+  }
+
+ private:
+  grpc_core::ResolverFactory* factory_;
+};
+
+}  // namespace
+
+TEST_F(BinderResolverTest, WrongScheme) {
+  TestFails("bonder:10.2.1.1");
+  TestFails("http:google.com");
+}
+
+// Authority is not allowed
+TEST_F(BinderResolverTest, AuthorityPresents) {
+  TestFails("binder://example");
+  TestFails("binder://google.com");
+  TestFails("binder://google.com/test");
+}
+
+// Path cannot be empty
+TEST_F(BinderResolverTest, EmptyPath) {
+  TestFails("binder:");
+  TestFails("binder:/");
+  TestFails("binder://");
+}
+
+TEST_F(BinderResolverTest, PathLength) {
+  // Assert the path length size is at least 101 so it can store a 100 character
+  // long path with a null terminator. (Typically the limit 108 on Linux and
+  // 104 on MacOS)
+  ASSERT_GE(sizeof(sockaddr_un::sun_path), 101);
```

Comment:
This doesn't really add any value, since you have a static assert for this in the code.  It won't build if this is not true, so when this test runs, this assertion will always be true.

Reply:
Removed